src/Utilities/PaymentHelpers.res

```diff
@@ -380,6 +446,61 @@ let rec intentCall = (
                   ])
                 }
                 resolve(data)
+              } else if intent.nextAction.type_ === "three_ds_invoke" {
+                let threeDsData =
+                  intent.nextAction.three_ds_data
+                  ->Belt.Option.flatMap(Js.Json.decodeObject)
+                  ->Belt.Option.getWithDefault(Js.Dict.empty())
+                let do3dsMethodCall =
+                  threeDsData
+                  ->Js.Dict.get("three_ds_method_details")
+                  ->Belt.Option.flatMap(Js.Json.decodeObject)
+                  ->Belt.Option.flatMap(x => x->Js.Dict.get("three_ds_method_data_submission"))
+                  ->Belt.Option.getWithDefault(Js.Dict.empty()->Js.Json.object_)
+                  ->Js.Json.decodeBoolean
+                  ->Utils.getBoolValue
+
+                let headerObj = Js.Dict.empty()
+                headers->Js.Array2.forEach(
+                  entries => {
+                    let (x, val) = entries
+                    Js.Dict.set(headerObj, x, val->Js.Json.string)
+                  },
+                )
+                Js.log2("HERE IS DATA", threeDsData)
+                let metaData =
+                  [
+                    ("threeDSData", threeDsData->Js.Json.object_),
+                    ("paymentIntentId", clientSecret->Js.Json.string),
+                    ("headers", headerObj->Js.Json.object_),
+                    ("url", url.href->Js.Json.string),
+                    ("iframeId", iframeId->Js.Json.string),
+                  ]->Js.Dict.fromArray
+
+                // handleLogging(
```

Comment:
Why is this code commented?

Reply:
we need to add logging right this will be helpful